api/feature_api.rb

```
@@ -1,25 +1,138 @@
+        feature_exist!
+        feature = current_feature
+
+        user_id = params[:user_id]
+        response = feature.add_user(user_id)
+
+        RestfulModels::Response.represent(message: response)
+      end
+
+      delete '/' do
+        feature_exist!
+        user_id = params[:user_id]
+        feature = current_feature
+
+        response = feature.remove_user(user_id)
+
+        status 200
+        RestfulModels::Response.represent(message: response)
+      end
+    end
+
+    get '/' do
+      feature_exist!
+      feature = current_feature
+
+      if feature.valid?
+        RestfulModels::Response.represent(data: RestfulModels::Feature.represent(feature))
+      else
+        status 500
+        RestfulModels::Response.represent(message: 'Error, feature is not valid')
+      end
+    end
+
+    params do
+      requires :user_id, type: Integer, desc: 'The user ID'
+    end
+    get '/:user_id/active' do
+      feature_exist!
+
+      user_id = params[:user_id].to_i
+      feature = current_feature
+
+      active = feature.active?(user_id: user_id)
+
+      RestfulModels::Response.represent(data: { active: active })
+    end
+
+    delete '/' do
+      feature_exist!
+      feature = current_feature
+
+      feature.delete
+      RestfulModels::Response.represent(message: 'The feature has been removed.')
+    end
+
+    params do
+      requires :description, type: String, desc: 'The feature description'
+      requires :author, type: String, desc: 'The author name'
+    end
+    post '/' do
+      options = {
+          name: params[:name],
+          percentage: params[:percentage] || 0,
+          description:  params[:description],
+          author: params[:author],
+          created_at: Time.current,
+          created_by: params[:author]
+      }
+
+      feature = Feature.new(options)
+
+      begin
+        feature.save!
+        RestfulModels::Response.represent(message: 'Feature created successfully!')
+      rescue Exception
```

i think `rescue => e` will catch all exceptions that you want to catch. 
http://daniel.fone.net.nz/blog/2013/05/28/why-you-should-never-rescue-exception-in-ruby/
(how about sharing the exception on the message ? we wouldn't like to share it?)

---

Great!

---

The more-standard way (as we do in other chimeras) is to handle grape exceptions using ```rescue_all` - this will save you the rescue block for each route.
Example: https://github.com/fiverr/mobile_chimera/blob/master/service/mobile_chimera.rb#L24

---

@ramigluzman 
I'll add it in addition - as a failsafe